state-chain/pallets/cf-swapping/src/tests.rs

```diff
@@ -2075,3 +2077,156 @@ fn can_swap_ccm_below_max_amount() {
+		});
+	});
+}
+
+#[test]
+fn swap_broker_fee_cannot_exceed_amount() {
+	new_test_ext().execute_with(|| {
+		const ALICE: u64 = 2_u64;
+		swap_with_custom_broker_fee(Asset::Usdc, Asset::Flip, 100, 15000);
+		assert_eq!(EarnedBrokerFees::<Test>::get(ALICE, cf_primitives::Asset::Usdc), 100);
+	});
+}
+
+fn swap_scheduled_event_witnessed(
+	swap_id: u64,
+	source_asset: Asset,
+	deposit_amount: AssetAmount,
+	destination_asset: Asset,
+	broker_commission: Option<AssetAmount>,
+) {
+	System::assert_last_event(RuntimeEvent::Swapping(Event::<Test>::SwapScheduled {
+		swap_id,
+		source_asset,
+		deposit_amount,
+		destination_asset,
+		destination_address: EncodedAddress::Eth([2; 20]),
+		origin: SwapOrigin::DepositChannel {
+			deposit_address: EncodedAddress::Eth([2; 20]),
+			channel_id: 1,
+			deposit_block_height: Default::default(),
+		},
+		swap_type: SwapType::Swap(ForeignChainAddress::Eth([2; 20].into())),
+		broker_commission,
+	}));
+}
+#[test]
+fn swap_broker_fee_subtracted_from_swap_amount() {
+	new_test_ext().execute_with(|| {
+		const ALICE: u64 = 2_u64;
+		let amounts: [AssetAmount; 6] = [50, 100, 200, 500, 1000, 10000];
+		let fees: [BasisPoints; 4] = [100, 1000, 5000, 10000];
+		const BASIS_POINTS_PER_MILLION: u32 = 100;
+
+		let combinations = amounts.iter().cartesian_product(fees);
+		let mut swap_id = 1;
+		Asset::all().iter().for_each(|asset| {
+			let mut total_fees = 0;
+			combinations.clone().for_each(|(amount, fee)| {
+				swap_with_custom_broker_fee(*asset, Asset::Flip, *amount, fee);
+				let broker_commission =
+					Permill::from_parts(fee as u32 * BASIS_POINTS_PER_MILLION) * *amount;
+				total_fees += broker_commission;
+				assert_eq!(EarnedBrokerFees::<Test>::get(ALICE, *asset), total_fees);
+				swap_scheduled_event_witnessed(
+					swap_id,
+					*asset,
+					*amount,
+					Asset::Flip,
+					Some(broker_commission),
+				);
+				swap_id += 1;
+			})
+		});
+	});
+}
+
+fn swap_amount_too_low_witnessed(asset: Asset, amount: AssetAmount) {
+	System::assert_last_event(RuntimeEvent::Swapping(Event::<Test>::SwapAmountTooLow {
+		asset,
+		amount,
+		destination_address: EncodedAddress::Eth([2; 20]),
+		origin: SwapOrigin::DepositChannel {
+			deposit_address: EncodedAddress::Eth([2; 20]),
+			channel_id: 1,
+			deposit_block_height: Default::default(),
+		},
+	}));
+}
+#[test]
+fn swap_fail_if_below_minimum_swap_amount() {
+	new_test_ext().execute_with(|| {
+		let asset = Asset::Usdc;
+		let amount = 1_000u128;
+		assert_eq!(MinimumSwapAmount::<Test>::get(asset), 0);
+
+		// Set the new minimum swap_amount (1000 USDC)
+		assert_ok!(Swapping::set_minimum_swap_amount(RuntimeOrigin::root(), asset, amount));
+
+		assert_eq!(MinimumSwapAmount::<Test>::get(asset), amount);
+		assert_eq!(Swapping::minimum_swap_amount(asset), amount);
+
+		System::assert_last_event(RuntimeEvent::Swapping(Event::<Test>::MinimumSwapAmountSet {
+			asset,
+			amount,
+		}));
+
+		const ALICE: u64 = 2_u64;
```

Comment:
We can just pull the const out of the `swap_with_custom_broker_fee` and then use it here - they need to be linked, so we should create that link explicitly by using the same const

same in the other place/s we do this too ofc

Reply:
My bad here we already have a global ALICE we can use for the tests